trpcultivate_phenotypes/tests/src/Kernel/ServiceGenusProjectTest.php

```diff
@@ -97,27 +156,46 @@ protected function setUp(): void {
    * @return array
    *   Each genus-project test scenario is an array witht the following values:
    *   - A string, human-readable short description of the test scenario.
-   *   - A string, the name or title of a project.
-   *   - An array, the genus and species of the organism that will be
-   *     paired with the project. Keyed by 'genus' and 'species'.
+   *   - An integer indicating the number of genus to set to the project.
    *   - An array of expected values, with the following keys:
    *     - 'project_genus': the expected genus returned by the method
    *     getGenusOfProject().
-   *     - 'alternative_genus': a genus returned after setting a genus.
    */
   public function provideGenusProjectForGenusProjectService() {
```

Comment:
It looks like we are missing the case where we try to set a genus for a project which is already set. For example, if the genus `Tripalus` is already configured for `Project A` and then we call `setGenusToProject('Project A', 'Tripalus)`. This should just return TRUE without changing the prop record that already exists.

This is also needed to cover line 105 in the service:
```php
        // Determine if the project already had the genus.
	        foreach ($project_genus as $row) {
	          if ($row->value == $genus) {
	            $project_has_genus = TRUE;
	            $break;
	          }
	        }
```

NOTE: When copying the above code sample I also spotted another bug. It should be `break;` not `$break;`.